.github/docs.yml

```
@@ -0,0 +1,96 @@
+#
+# Build documentation and deploy
+#
+name: Docs
+
+on:
+  # pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docs/**'
+      - '.github/workflows/**'
+
+defaults:
+  run:
+    # default to use bash shell
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      # Cancel previous runs that are not completed
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v2.4.0
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install requirements
+        run: pip install -r requirements.txt
+
+      - name: Build the website
+        run: make -C docs clean html
+
+      - name: Checkout the gh-pages branch
+        uses: actions/checkout@v2.4.0
+        with:
+          ref: gh-pages
+          # Checkout to this folder instead of the current one
+          path: deploy
+          # Download the entire history
+          fetch-depth: 0
+        if: (github.event_name == 'push') && (matrix.os == 'ubuntu-latest')
```

No need to check `matrix.os` here.
```suggestion
        if: github.event_name == 'push'
```